src/generic/stage2/numberparsing.h

```diff
@@ -224,39 +198,16 @@ simdjson_really_inline bool compute_float_64(int64_t power, uint64_t i, bool neg
   return true;
 }
 
-static bool parse_float_strtod(const uint8_t *ptr, double *outDouble) {
-  char *endptr;
-  // We want to call strtod with the C (default) locale to avoid
-  // potential issues in case someone has a different locale.
-  // Unfortunately, Visual Studio has a different syntax.
-#ifdef _WIN32
-  static _locale_t c_locale = _create_locale(LC_ALL, "C");
-  *outDouble = _strtod_l((const char *)ptr, &endptr, c_locale);
-#else
-  static locale_t c_locale = newlocale(LC_ALL_MASK, "C", NULL);
-  *outDouble = strtod_l((const char *)ptr, &endptr, c_locale);
-#endif
-  // Some libraries will set errno = ERANGE when the value is subnormal,
-  // yet we may want to be able to parse subnormal values.
-  // However, we do not want to tolerate NAN or infinite values.
-  //
-  // Values like infinity or NaN are not allowed in the JSON specification.
-  // If you consume a large value and you map it to "infinity", you will no
-  // longer be able to serialize back a standard-compliant JSON. And there is
-  // no realistic application where you might need values so large than they
-  // can't fit in binary64. The maximal value is about  1.7976931348623157 x
-  // 10^308 It is an unimaginable large number. There will never be any piece of
-  // engineering involving as many as 10^308 parts. It is estimated that there
-  // are about 10^80 atoms in the universe.  The estimate for the total number
-  // of electrons is similar. Using a double-precision floating-point value, we
-  // can represent easily the number of atoms in the universe. We could  also
-  // represent the number of ways you can pick any three individual atoms at
-  // random in the universe. If you ever encounter a number much larger than
-  // 10^308, you know that you have a bug. RapidJSON will reject a document with
-  // a float that does not fit in binary64. JSON for Modern C++ (nlohmann/json)
-  // will flat out throw an exception.
-  //
-  if ((endptr == (const char *)ptr) || (!std::isfinite(*outDouble))) {
+// We call a fallback floating-point parser that might be slow. Note
```

Comment:
Ah, so you were able to remove if (endptr == ptr) because we already know there is at least one digit, due to having tried to parse it the normal way! Nice.

Reply:
Adding a remark. Yes.